js/controller.js

```diff
@@ -7,20 +7,26 @@ class Controller {
         this.mouseDown = false;
         this.mouseDownX = 0;
         this.mouseDownY = 0;
+        this.mouse = {
+            x: 0,
+            y: 0
+        };
     }
 
     keyListener = function (event) {
-        if (event.type == "mousedown") {
+        if (event.type === "mousedown") {
             this.mouseDown = true;
             this.mouseDownX = event.offsetX;
             this.mouseDownY = event.offsetY;
-        }
-        else if (event.type == "mouseup") {
+        } else if (event.type === "mouseup") {
             this.mouseDown = false;
             this.mouseDownX = 0;
             this.mouseDownY = 0;
-        }
-        else { }
+        } else {}
     }
 
+    updateMousePos(event) {
```

Comment:
good idea separating this into a separate method. we should do that with the others as well